scripts/dashboard-esbuild.ts

```diff
@@ -0,0 +1,358 @@
+const packageJSON = JSON.parse(fs.readFileSync(join(__dirname, '../package.json'), 'utf8'))
+const CONTRACTS_VERSION = packageJSON.contractsVersion || '2.0.0'
+const CHEL_VERSION = packageJSON.version
+
+// Fix for __proto__ issue with buble (used by @vue/component-compiler)
+// See: https://github.com/denoland/deno/issues/20618
+if (({} as Record<PropertyKey, unknown>).__proto__ !== Object.prototype) {
+  Object.defineProperty(Object.prototype, '__proto__', {
+    get () {
+      return Object.getPrototypeOf(this)
+    },
+    set (v) {
+      Object.setPrototypeOf(this, v)
+    }
+  })
+}
+
+// Alias replacer function (adapted from Group Income's utils.js)
+function createAliasReplacer (aliases: Record<string, string>) {
+  if (Object.keys(aliases).some(alias => alias.includes('/'))) {
+    throw new Error('Path aliases may not include slash characters.')
+  }
+
+  const cwd = process.cwd()
+  const escapeForRegExp = (string: string) => string.replace(/[.*+?^${}()|\\[\]]/g, '\\$&')
+  const escapedAndSortedAliases = Object.keys(aliases).map(escapeForRegExp).sort().reverse()
+  const re = new RegExp(
+    `(?:^import[ (]|\\bimport[ (]|import .+? from |^\\} from )['"](${ escapedAndSortedAliases.join('|')})(?:['"]|/[^'"]+?['"])`,
+    'gm'
+  )
+
+  return function aliasReplacer ({ path, source }: { path: string, source: string }) {
+    const relativeDirPath = relative(dirname(path), cwd)
+
+    return source.replace(re, (match, capture) => {
+      const resolvedPathSegment = aliases[capture]
+      const replacement = join(relativeDirPath, resolvedPathSegment)
+        .replace(/\\/g, '/')
+
+      return match.replace(capture, replacement)
+    })
+  }
+}
+
+// Copy index.html to output directory
+async function copyIndexHtml (outDir: string) {
+  try {
+    const sourceHtml = join(__dirname, '../src/serve/dashboard/index.html')
+    const targetHtml = join(outDir, 'index.html')
+
+    // Read the source HTML
+    const htmlContent = await readFile(sourceHtml, 'utf8')
+
+    // Write the HTML to output directory
+    await writeFile(targetHtml, htmlContent, 'utf8')
+    console.log('📄 index.html copied')
+  } catch (error) {
+    console.error('❌ Failed to copy index.html:', error)
+    throw error
+  }
+}
+
+// Extract CSS from Vue components and create combined-styles.scss
+export async function extractAndCreateCSS (outDir: string) {
+  try {
+    const vueFiles = await findVueFiles(dashboardDir)
+    let combinedCSS = ''
+
+    for (const vueFile of vueFiles) {
+      const source = await readFile(vueFile, 'utf-8')
+      const styleMatch = source.match(/<style[^>]*scoped[^>]*>([\s\S]*?)<\/style>/)
+
+      if (styleMatch) {
+        const componentName = basename(vueFile, '.vue')
+        combinedCSS += `\n/* Styles from ${componentName}.vue */\n${styleMatch[1]}\n`
+      }
+    }
+
+    if (combinedCSS) {
+      const cssDir = join(outDir, 'assets', 'css')
+      await mkdir(cssDir, { recursive: true })
```

Comment:
I don't understand why you're doing this manually as esbuild is supposed to handle this automatically with any Vuejs plugins for esbuild.

Can you see if you can get it work as it's intended with esbuild + vue compiler?

If not can you please explain what you tried and what problems you ran into?

Reply:
I thought this was no longer used.

Anyhow, I noticed (this seems to go for GI as well) that the Vue build process isn't as good as it could be. If you look at the files being output, you'll see a lot of code that should be compiled isn't (for example, styles and pug templates). I think that we should look into improving the build process since pug shouldn't be a runtime dependency, and the styles, well, SCSS shouldn't be runtime dependency either (also, probably SCSS is no longer needed, since most of what it does is now natively supported by CSS).